core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -579,15 +579,22 @@ class LogManager(logDirs: Seq[File],
 
   /**
    * Make a checkpoint for all logs in provided directory.
+   * The logOptOfTopicPartition argument is an optimization that can be used if we only want to checkpoint a given partition.
    */
-  private def checkpointLogRecoveryOffsetsInDir(dir: File): Unit = {
+  private def checkpointLogRecoveryOffsetsInDir(dir: File, logOptOfTopicPartition: Option[Log] = None): Unit = {
```

Comment:
A few suggestions:
1. Maybe pass a `Option[Seq[Log]]` here, remove the default argument and call it `affectedLogs`.
2. In the documentation, we should specify that if a subset of logs are passed, we can optimize the producer snapshot deletion process. The actual checkpointing always involves all the logs.